insights/parsers/ip_netns_exec_namespace_lsof.py

```diff
@@ -0,0 +1,82 @@
+"""
+IpNetnsExecNamespaceLsofI - command ``/sbin/ip netns exec [network-namespace] lsof -i"``
+========================================================================================
+
+This module provides class ``IpNetnsExecNamespaceLsofI`` for parsing the output of command
+``/sbin/ip netns exec [network-namespace] lsof -i``.
+Filters have been added so that sensitive information can be filtered out.
+This results in the modification of the original structure of data.
+"""
+
+from collections import namedtuple
+from insights import CommandParser, parser
+from insights.core.filters import add_filter
+from insights.parsers import SkipException, keyword_search, parse_fixed_table
+from insights.specs import Specs
+
+FILTERS = ["COMMAND", "neutron-n"]
+add_filter(Specs.ip_netns_exec_namespace_lsof, FILTERS)
+
+
+@parser(Specs.ip_netns_exec_namespace_lsof)
+class IpNetnsExecNamespaceLsofI(CommandParser):
+    """
+       This class provides processing for the output of the command
+       ``/sbin/ip netns exec [network-namespace] lsof -i``.
+
+       Sample command output::
+
+           COMMAND   PID   USER    FD  TYPE  DEVICE     SIZE/OFF  NODE NAME
+           neutron-n 975   root    5u  IPv4  6482691    0t0        TCP *:http (LISTEN)
+
+       Examples:
+           >>> len(data.search(command="neutron-n"))
+           1
+           >>> data.data[0]["command"] == "neutron-n"
+           True
+
+       Attributes:
+           fields (list): List of ``KeyValue`` namedtupules for each line
+                          in the command.
+
+           data (list): List of key value pair derived from the command.
+
+       Raises:
+           SkipException: When the file is empty or data is useless.
+    """
+    keyvalue = namedtuple("KeyValue",
+                          ["command", "pid", "user", "fd", "type", "device", "size_off", "node", "name"])
+
+    def parse_content(self, content):
+        self.fields = []
+        self.data = []
+        if not content:
+            raise SkipException("Empty file")
+
+        self.data = parse_fixed_table(content,
+                                      heading_ignore=["COMMAND", "PID", "USER", "FD", "TYPE", "DEVICE", "SIZE/OFF", "NODE", "NAME"],
+                                      header_substitute=[("COMMAND", "command"), ("PID", "pid"), ("USER", "user"), ("FD", "fd"),
+                                                         ("TYPE", "type"), ("DEVICE", "device"), ("SIZE/OFF", "size_off"),
+                                                         ("NODE", "node"), ("NAME", "name")])
+
+        for item in self.data:
+            self.fields.append(self.keyvalue(item["command"], item["pid"], item["user"], item["fd"], item["type"], item["device"],
+                               item["size_off"], item["node"], item["name"]))
+
+        if not self.data:
```

Comment:
Better move this check ahead of the for-clause at lin-62

Reply:
@xiangce - Acknowledged. Changes done in the latest commit.